File: Backend/AstroGame.Shared/Models/Researches/ResearchCost.cs
using AstroGame.Shared.Models.Resources;
using System;

namespace AstroGame.Shared.Models.Researches
{
    public abstract class ResearchCost
    {
        public Guid Id { get; set; }
        public Guid ResearchId { get; set; }
        public Guid ResourceId { get; set; }

        public Research Research { get; set; }
        public Resource Resource { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/researches/research_cost.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/communications/converters/research_cost_converter.dart';
import 'package:flutter_guid/flutter_guid.dart';
import 'package:json_annotation/json_annotation.dart';

@GuidConverter()
@JsonSerializable()
abstract class ResearchCost {
  Guid id;
  Guid researchId;
  Guid resourceId;

  ResearchCost();

  factory ResearchCost.fromJson(Map<String, dynamic> json) => ResearchCostConverter().fromJson(json);
}
